src/libpriv/rpmostree-core.c

```diff
@@ -747,18 +751,18 @@ rpmostree_context_setup (RpmOstreeContext    *self,
   if (!dnf_context_setup (self->dnfctx, cancellable, error))
     return FALSE;
 
-  /* disable all repos in pkgcache-only mode, otherwise obey "repos" key */
-  if (self->pkgcache_only)
-    {
-      if (!context_repos_enable_only (self, NULL, error))
-        return FALSE;
-    }
-  else
+  /* start with all repos off, then only turn on the ones specified in the treefile */
+  disable_all_repos (self);
+
+  /* all repos stay disabled in pkgcache-only mode */
+  if (!self->pkgcache_only)
     {
       /* NB: missing "repos" --> let hif figure it out for itself */
-      if (g_variant_dict_lookup (self->spec->dict, "repos", "^a&s", &enabled_repos))
-        if (!context_repos_enable_only (self, (const char *const*)enabled_repos, error))
-          return FALSE;
+      { g_autofree char **enabled_repos = NULL;
+        if (g_variant_dict_lookup (self->spec->dict, "repos", "^a&s", &enabled_repos))
```

Comment:
Let's use `{` for nested conditionals; it's more robust to future refactorings.

Reply:
Yup, just moving existing code around. Though actually, I can also just drop the extra scoping I added here. It was initially prep for https://github.com/coreos/rpm-ostree/pull/2058 but meh I can easily just use a different variable name.